test/promise.js

```diff
@@ -191,6 +191,62 @@ describe('Promise', function() {
       assert(prom instanceof SubClass);
     });
   });
+  describe('Promise.prototype.finally', function() {
+    it('should be called', function(done) {
+      var called = false;
+      Promise.resolve().finally(function() {
+        called = true;
+      });
+      setTimeout(function() {
+        assert(called, 'Finally handler was called');
+        done();
+      }, 50);
+    });
+
+    it('should be called on success', function(done) {
+      Promise.resolve(3).finally(function() {
+        assert(true, 'Finally handler was called');
```

Comment:
This assert is unnecessary. It never fails. Can you change this to

```js
Promise.resolve(3).finally(function() {
  assert.equal(arguments.length, 0,  'No arguments to onFinally');
  done();
});
```

Reply:
Actually it would fail if it was never called (by a timeout in the testing framework). But I can check there are no args provided. Should also add a new test verifying that if the handler returns a promise, it must be awaited for.